### DISCUSSION TITLE
smaller version

### DISCUSSION THREAD

**Discussion by namork** [gkoh/furble#142] [General]
would it be possible to build a version of furble on a tiny esp32 without display to see better battery life?

for example... https://shop.m5stack.com/products/m5stack-nanoc6-dev-kit

**Comment by matthudsonau** [comment]
You'd need a way to pair it to the camera, and without a screen that's going to be a challenge. If that problem can be solved (maybe a configuration mode via USB?) then it should be at least possible, although the behaviour would need to be changed to autoconnect on power on and disconnect

**Comment by gkoh** [comment]
@matthudsonau Agreed, no display is a challenge. Unlike most commercial remotes, furble is designed for multi-target use. But yes, we could distill a version into a singular purpose package.

@namork Is your actual request related to size or battery life?
A smaller unit ultimately results in a smaller battery.
With the M5Stack units, the main power hog is usually the esp32 itself not the display. The esp32 is primarily designed for WiFi which is tied to the Bluetooth hardware and eats power.

Having said that, I am optimistic I can improve battery on existing units. However I want to finish the UI rewrite first. I want to tweak the hardware configuration which requires moving away from Arduino as the base framework.

**Comment by gkoh** [comment]
@namork With v3.0.0 just released, I have migrated the underlying platform.
This has exposed much more power management control for the ESP32.

I expect v3.0.0 to exhibit measurable improvements in battery life.
See here if you want to see some random testing I did: [Power Consumption](https://github.com/gkoh/furble/wiki/Power-Consumption)

(Just don't use GPS, which appears to chew through power.)

**Comment by namork** [comment]
thank you very much! did you ever test the software on an m5core ink? it has a 390 mA battery and an e-ink display…

**Reply by gkoh** [reply]
> thank you very much!

You are very welcome!

> did you ever test the software on an m5core ink? it has a 390 mA battery and an e-ink display…

This was (very) briefly considered quite some time ago. At the time the Core Ink was significantly more expensive (compared to the original $USD9.95 M5StickC) and always seemed out of stock.
With the newest M5StickC Plus2 doubling the entry price, the difference is much less 'shocking'.

If there is interest and you are willing to help test, I am OK to purchase and support.

Just running some random numbers in my head, I would guess that (without GPS unit):
* 390 mAh battery = ~2x life time
* e-ink screen will reduce power compared to M5StickC Plus2 by 80%
* = ~3x overall improvement on battery compared to M5StickC Plus2

Let me know if you would like to see this.

**Reply by namork** [reply]
well gkoh, are you sure? i think it only makes sense if this investment would be interesting for others, too. the m5core unfortunately is a little bit bulky. but would definitely buy one, if it is useful as grx100 remote and i don't know of an other esp32 with e-ink screen in housing.

**Reply by gkoh** [reply]
Actually, I'm not sure.
I do like the idea of the e-ink display and generally lower power consumption and improved battery.
However, you're right, the Core Ink:
* is a bit bulkier (not really a concern to me)
* has a different control mechanism, that interesting 3-way switch
* seems much less popular and less available
* was released some years ago

The age and popularity of the device are what concerns me most, M5Stack seem to deprecate/obsolete things all the time.
Perhaps the right answer is to wait a bit, so let's do that for a bit and keep an eye on the market.

The closest I have found is this, but ... no battery:
* [LILYGO Mini E-Paper](https://lilygo.cc/products/mini-e-paper-core)

Everything else is either:
* too big
* too expensive
* no tactile buttons

**Reply by namork** [reply]
gkoh. the 3-way switch is actually very nice. i had the m5core ink in my hands. it is also nice to have a buzzer build in. but i agree, it should be a newer esp32 and for this use case it would be nice the device would be smaller. it might be also useful to have it in form of a watch. so let's see... thanks for your support.

**Comment by gkoh** [comment]
@namork Thank you for the information, useful.
There isn't a strict need to restrict ourselves to just ESP32, I myself have been looking into nRF52 targets. It would be a different project, but porting the camera protocols is actually straightforward.

LILYGO actually had an nRF52 wristband that I wanted to acquire:
* https://github.com/Xinyuan-LilyGO/LilyGo-T-Wristband-NRF52

Sadly, it wasn't on the market very long and was discontinued.

We'll keep looking!
